packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -352,15 +377,31 @@ export function jsxProdSignatureRunningInDevWithDynamicChildren(
     const trackActualOwner =
       __DEV__ &&
       ReactSharedInternals.recentlyCreatedOwnerStacks++ < ownerStackLimit;
+    let debugStackDEV = false;
+    if (__DEV__) {
+      if (trackActualOwner) {
+        if (supportsSettingStackTraceLimit) {
```

Comment:
I'm concerned about this unnecessary condition adding to code size (in terms of memory / JIT optimization and not download size) and the extra lookup every call.

We should make two entire copies of the function and switch which function we use instead so that the function that we use doesn't have to make any additional checks at runtime.

Alternatively we should just assume that you can set it and not support environments that don't play ball.